internal/services/servicebus/servicebus_namespace_resource.go

```diff
@@ -250,3 +266,63 @@ func resourceServiceBusNamespaceDelete(d *pluginsdk.ResourceData, meta interface
 
 	return nil
 }
+
+func expandServiceBusNamespaceIdentity(input []interface{}) (*servicebus.Identity, error) {
+	if len(input) == 0 || input[0] == nil {
+		return &servicebus.Identity{
+			Type: servicebus.ManagedServiceIdentityTypeNone,
+		}, nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	config := &servicebus.Identity{
+		Type: servicebus.ManagedServiceIdentityType(v["type"].(string)),
+	}
+
+	identityIds := v["identity_ids"].(*pluginsdk.Set).List()
+
+	if len(identityIds) != 0 {
+		if config.Type != servicebus.ManagedServiceIdentityTypeSystemAssignedUserAssigned && config.Type != servicebus.ManagedServiceIdentityTypeUserAssigned {
+			return nil, fmt.Errorf("`identity_ids` can only be specified when `type` includes `UserAssigned`")
+		}
+		config.UserAssignedIdentities = map[string]*servicebus.UserAssignedIdentity{}
+		for _, id := range identityIds {
+			config.UserAssignedIdentities[id.(string)] = &servicebus.UserAssignedIdentity{}
+		}
+	}
+
+	return config, nil
+}
+
+func flattenServiceBusNamespaceIdentity(input *servicebus.Identity) ([]interface{}, error) {
+	if input == nil || input.Type == servicebus.ManagedServiceIdentityTypeNone {
+		return []interface{}{}, nil
+	}
+
+	coalesce := func(input *string) string {
+		if input == nil {
+			return ""
+		}
+
+		return *input
+	}
+
+	var identityIds []string
+	for id := range input.UserAssignedIdentities {
+		parsedId, err := msiParse.UserAssignedIdentityIDInsensitively(id)
+		if err != nil {
+			return nil, err
+		}
+		identityIds = append(identityIds, parsedId.ID())
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"type":         input.Type,
+			"identity_ids": identityIds,
+			"principal_id": coalesce(input.PrincipalID),
+			"tenant_id":    coalesce(input.TenantID),
+		},
+	}, nil
```

Comment:
this should be using the [commonschema](https://github.com/hashicorp/go-azure-helpers/blob/main/resourcemanager/commonschema/identity_system_user.go) and [expand/flatten functions](https://github.com/hashicorp/go-azure-helpers/blob/main/resourcemanager/identity/system_and_user_assigned_map.go) from hashicorp/go-azure-helpers (see #15187 for more info)